content/Software and Downloads/Compilation/If-your-sketch-doesnt-compile.md

```diff
@@ -0,0 +1,242 @@
+   * Compilation errors will often reference the row and column where the error was triggered, i.g. `Blink:29:1` (line 29, 1 column).
+   * You may have to scroll or expand the window width to see the entire message.
+
+3. Look for highlights in the editor:
+   * In IDE 1.x, the **line** where the error occurred is highlighted in red.
+   * In IDE 2.x, the **character** where the error occurred is underlined in red.
+   * The line causing the error may be before the line where it was triggered. In the example above, line 28 is missing a semicolon (`;`), but the error will be triggered by the unexpected closing bracket (`}`) in line 29.
+
+   <table class="img-captions">
+     <tr>
+        <td><img src="img/ide1-syntax-error-output-highlighted.png" alt="Compilation error in IDE 1.8.13, due to a missing semicolon."></td>
+        <td><img src="img/ide2-syntax-error-output-highlighted.png" alt="Compilation error in IDE 2.0.0-3c6, due to a missing semicolon."></td>
+     </tr>
+     <tr>
+        <td>Error output in IDE 1.x.</td>
+        <td>Error output in IDE 2.x.</td>
+     </tr>
+   </table>
+
+---
+
+## Troubleshoot a specific error
+
+<table>
+<tr>
+<td style="vertical-align: top;">
+<h3>Common</h3>
+<li><a href="#error-compiling-for-board">Error compiling for board</a></li>
+<li><a href="#missing-FQBN">Compilation error: Missing FQBN (Fully Qualified Board Name)</a></li>
+<li><a href="#platform-not-installed">Compilation error: Error: 2 UNKNOWN: platform not installed</a></li>
+<h3>Required functions</h3>
+<ul>
+<li><a href="#required-functions">undefined reference to 'setup'</a></li>
+<li><a href="#required-functions">undefined reference to 'loop'</a></li>
+<li><a href="#required-functions">collect2: error: ld returned 1 exit status</a></li>
+</ul>
+<h3>Libraries</h3>
+<ul>
+<li><a href="#fatal-error-no-such-file-or-directory">fatal error: : No such file or directory</a></li>
+</ul>
+</td>
+<td style="vertical-align: top;">
+<h3>Syntax errors</h3>
+<ul>
+<li><a href="#">variable was not declared in this scope</a></li>
+<li><a href="#">expected unqualified-id before numeric constant</a></li>
+<li><a href="#">expected ',' or ';' before ':' token</a></li>
+<li><a href="#"></a></li>
+<li><a href="#"></a></li>
+<li><a href="#"></a></li>
+<li><a href="#"></a></li>
+</ul>
+</td>
+</tr>
+</table>
+
+<!-- COMMON -->
+
+<a id="missing-FQBN"></a>
+
+### Compilation error: Missing FQBN (Fully Qualified Board Name)
+
+When compiling or uploading code, this error will occur if no board is selected, or if the board core is not installed.
+
+A board needs to be selected even if you're not uploading any code.
+
+<a id="platform-not-installed"></a>
+
+### Compilation error: Error: 2 UNKNOWN: platform not installed
+
+This error occurs when the required *board core* is not installed. Please ensure the core is installed using [this guide](https://www.arduino.cc/en/Tutorial/getting-started-with-ide-v2/ide-v2-board-manager#installing-a-core).
```

Comment:
The link directs to the main IDE 2 tutorials, not to the installing a core guide

Reply:
6134b5e